plugin/trino-postgresql/src/test/java/io/trino/plugin/postgresql/TestTestingPostgreSqlServer.java

```diff
@@ -25,19 +22,20 @@
 import java.sql.DriverManager;
 import java.sql.SQLException;
 import java.sql.Statement;
-import java.util.List;
-import java.util.Set;
 import java.util.concurrent.ExecutorService;
 import java.util.concurrent.Future;
 
 import static io.airlift.concurrent.Threads.daemonThreadsNamed;
 import static io.airlift.testing.Closeables.closeAll;
 import static io.trino.plugin.jdbc.RemoteDatabaseEvent.Status.CANCELLED;
 import static io.trino.plugin.jdbc.RemoteDatabaseEvent.Status.RUNNING;
+import static io.trino.testing.assertions.Assert.assertEventually;
 import static java.util.concurrent.Executors.newCachedThreadPool;
 import static org.assertj.core.api.Assertions.assertThat;
 import static org.assertj.core.api.Assertions.assertThatThrownBy;
 
+// To make sure that we don't mix events between tests
+@Test(singleThreaded = true)
```

Comment:
Does it mean that anyone using testing events needs to be singlethreaded too?

BTW it seems the tests here are written in a way that doesn't require singlethreadedness

Reply:
> Does it mean that anyone using testing events needs to be singlethreaded too?

No. I just didn't want to assure that each test independent. However Maybe I was too defensive as you already pointed out _it seems the tests here are written in a way that doesn't require singlethreadedness_

Reply:
I don't care that much about this class being singlethreaded or not, but we use the same patterns elsewhere (in shared test class).
If we need all of them be singlethreaded, let's do it, but let's make sure all are marked as such.

Reply:
That annotation was only for this class.